modules/options/src/main/scala/scala/build/actionable/ActionableHandler.scala

```diff
@@ -0,0 +1,30 @@
+package scala.build.actionable
+
+import scala.build.errors.BuildException
+import scala.build.options.BuildOptions
+
+trait ActionableHandler {
+  type V
+
+  /** Extract options on the basis of which actionable diagnostics will be generated
+    *
+    * @param options
+    *   the Build Options to extract options
+    * @return
+    *   the list of options on the basis of which actionable diagnostics will be generated
+    */
+  def extractOptions(options: BuildOptions): Seq[V]
```

Comment:
I wonder if we really need to split the API into 2 parts: `extractOptions` and `createActionableDiagnostic`. I would personally stick to single method that accepts `BuildOptions` and `Inputs` and return `Either[BuildException, Seq[ActionableDiagnostic]]`

Reply:
From my side, splitting into two methods allows me, to implement the function `createActionableDiagnostics` and users don't have to implement logic to collect actionable diagnostic.